2005-01-09  Marcin Bohosiewicz <marcus at pld-linux.org>
	    Jakub Bogusz <qboosh at pld-linux.org>
	    Dmitry V. Levin <ldv at altlinux.org>

	* configure.ac: Link with -lnsl and -lsocket only when necessary.
	Merge two AC_CONFIG_FILES sections.

diff -urk.orig cpio-2.6.orig/configure.ac cpio-2.6/configure.ac
--- cpio-2.6.orig/configure.ac	2004-12-20 10:46:30 +0000
+++ cpio-2.6/configure.ac	2005-01-09 01:12:13 +0000
@@ -95,8 +95,8 @@
 AC_CHECK_FUNCS(lchown endpwent endgrent)
 AC_FUNC_VPRINTF
 AC_FUNC_ALLOCA
-AC_CHECK_LIB(nsl, gethostname, [LIBS="$LIBS -lnsl"])
-AC_CHECK_LIB(socket, setsockopt, [LIBS="$LIBS -lsocket"])
+AC_CHECK_FUNC(gethostname,,[AC_CHECK_LIB(nsl, gethostname, [LIBS="$LIBS -lnsl"])])
+AC_CHECK_FUNC(setsockopt,,[AC_CHECK_LIB(socket, setsockopt, [LIBS="$LIBS -lsocket"])])
 
 # Gettext.
 AM_GNU_GETTEXT([external], [need-ngettext])
@@ -104,10 +104,11 @@
 
 # Initialize the test suite.
 AC_CONFIG_TESTDIR(tests)
-AC_CONFIG_FILES([tests/Makefile tests/atlocal]) # FIXME: tests/preset?
 AM_MISSING_PROG([AUTOM4TE], [autom4te])
 
 AC_CONFIG_FILES([Makefile
+ tests/Makefile
+ tests/atlocal
  doc/Makefile
  headers/Makefile
  lib/Makefile
